### PR TITLE
Updates our fork to v6.6.0 from the upstream repo

### PR DESCRIPTION
**Pull request by shubhamparikh927**
- Updated the hapi-fhir to 6.6.0
- Tested locally by running the hapi fhir.
- Also verified the by running hapi fhir using docker.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,6 +4,9 @@ WORKDIR /usr/app//hapi-fhir-jpaserver-starter
 ARG OPENTELEMETRY_JAVA_AGENT_VERSION=1.17.0
 RUN curl -LSsO https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v${OPENTELEMETRY_JAVA_AGENT_VERSION}/opentelemetry-javaagent.jar
 
+ARG OPENTELEMETRY_JAVA_AGENT_VERSION=1.17.0
```
Comment (hankwallace) [inline Dockerfile]: Something went wrong here. These lines are duplicates from above.

File: pom.xml [hunk 2]
```diff
@@ -14,7 +14,7 @@
     <parent>
         <groupId>ca.uhn.hapi.fhir</groupId>
         <artifactId>hapi-fhir</artifactId>
-        <version>6.4.0</version>
+            <version>6.6.0</version>
```
Comment (hankwallace) [inline pom.xml]: The indentation here is wrong.

File: src/main/java/ca/uhn/fhir/jpa/starter/AppProperties.java [hunk 3]
```diff
@@ -556,6 +557,18 @@ public boolean getInstall_transitive_ig_dependencies() {
...
+	
+	public boolean getReload_existing_implementationguides() {
+		return reload_existing_implementationguides;
+	}
+	
+	public void setReload_existing_implementationguides(boolean reload_existing_implementationguides) {
+		this.reload_existing_implementationguides = reload_existing_implementationguides;
+	}
+
+	public List<String> getLocal_base_urls() {
```
Comment (hankwallace) [inline src/main/java/ca/uhn/fhir/jpa/starter/AppProperties.java]: This was merged at the wrong place. This same function was declared later in the upstream repo.

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java [hunk 4]
```diff
@@ -1,5 +1,37 @@
 package ca.uhn.fhir.jpa.starter.common;
 
+import static ca.uhn.fhir.jpa.starter.common.validation.IRepositoryValidationInterceptorFactory.ENABLE_REPOSITORY_VALIDATING_INTERCEPTOR;
```
Comment (hankwallace) [inline src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: Where did this whole block come from? I doesn't appear in the upstream repo.

Comment (hankwallace) [reply src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: This file appears to be very different than the upstream one. The only significant change we've made in our fork is to register interceptors when OAuth is enabled (lines ~ 443-449).

File: docker-compose.yml [hunk 5]
```diff
@@ -6,14 +6,6 @@ services:
     restart: on-failure
     ports:
       - "8080:8080"
-    environment:
```
Comment (hankwallace) [inline docker-compose.yml]: Why did you remove these lines?

File: src/test/smoketest/plain_server.http [hunk 6]
```diff
@@ -0,0 +1,223 @@
+### Create Single Patient
```
Comment (hankwallace) [inline src/test/smoketest/plain_server.http]: This appears to be a file rename. Shouldn't this pull request just rename `plain_server.rest`?

**Review by hankwallace — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 5 above)
Comment (shubhamparikh927) [reply docker-compose.yml]: I added these line again

File: Dockerfile (hunk 1 above)
Comment (hankwallace) [reply Dockerfile]: The line below this is still duplicated. The comment re: "these lines" was about 2 lines.

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java [hunk 7]
```diff
@@ -34,7 +65,13 @@
 import ca.uhn.fhir.jpa.packages.IPackageInstallerSvc;
 import ca.uhn.fhir.jpa.packages.PackageInstallationSpec;
 import ca.uhn.fhir.jpa.partition.PartitionManagementProvider;
-import ca.uhn.fhir.jpa.provider.*;
+import ca.uhn.fhir.jpa.provider.DaoRegistryResourceSupportedSvc;
```
Comment (hankwallace) [inline src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: This block does not match the upstream repo.

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java [hunk 8]
```diff
@@ -51,41 +88,30 @@
...
 import ca.uhn.fhir.jpa.validation.JpaValidationSupportChain;
-import ca.uhn.fhir.mdm.dao.IMdmLinkDao;
 import ca.uhn.fhir.mdm.provider.MdmProviderLoader;
 import ca.uhn.fhir.narrative.DefaultThymeleafNarrativeGenerator;
 import ca.uhn.fhir.narrative2.NullNarrativeGenerator;
 import ca.uhn.fhir.rest.api.IResourceSupportedSvc;
 import ca.uhn.fhir.rest.openapi.OpenApiInterceptor;
-import ca.uhn.fhir.rest.server.*;
-import ca.uhn.fhir.rest.server.interceptor.*;
+import ca.uhn.fhir.rest.server.ApacheProxyAddressStrategy;
```
Comment (hankwallace) [inline src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: This block does not match the upstream repo.

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java [hunk 9]
```diff
@@ -51,41 +88,30 @@
...
+import ca.uhn.fhir.rest.server.ETagSupportEnum;
+import ca.uhn.fhir.rest.server.HardcodedServerAddressStrategy;
+import ca.uhn.fhir.rest.server.IServerConformanceProvider;
+import ca.uhn.fhir.rest.server.IncomingRequestAddressStrategy;
+import ca.uhn.fhir.rest.server.RestfulServer;
+import ca.uhn.fhir.rest.server.interceptor.CorsInterceptor;
```
Comment (hankwallace) [inline src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: This block does not match the upstream repo. I'm not a fan of their use of `import .*`, but not matching their use in these blocks makes it harder to compare/update our fork. What's the reason to expand them?

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java [hunk 10]
```diff
@@ -51,41 +56,61 @@
...
 import ca.uhn.fhir.rest.server.interceptor.consent.ConsentInterceptor;
 import ca.uhn.fhir.rest.server.interceptor.partition.RequestTenantPartitionInterceptor;
 import ca.uhn.fhir.rest.server.provider.ResourceProviderFactory;
 import ca.uhn.fhir.rest.server.tenant.UrlBaseTenantIdentificationStrategy;
 import ca.uhn.fhir.rest.server.util.ISearchParamRegistry;
 import ca.uhn.fhir.validation.IValidatorModule;
 import ca.uhn.fhir.validation.ResultSeverityEnum;
-import com.google.common.base.Strings;
-import com.google.common.collect.ImmutableList;
+import static ca.uhn.fhir.jpa.starter.common.validation.IRepositoryValidationInterceptorFactory.ENABLE_REPOSITORY_VALIDATING_INTERCEPTOR;
```
Comment (hankwallace) [inline src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: This block does not match the upstream repo.

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java [hunk 11]
```diff
@@ -51,41 +56,61 @@
...
+import javax.sql.DataSource;
+
 import org.hl7.fhir.common.hapi.validation.support.CachingValidationSupport;
 import org.springframework.beans.factory.NoSuchBeanDefinitionException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.ApplicationContext;
-import org.springframework.context.annotation.*;
+import org.springframework.context.annotation.Bean;
```
Comment (hankwallace) [inline src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: This block does not match the upstream repo.

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java (hunk 10 above)
Comment (hankwallace) [reply src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: Please take another look at this file and fix ALL of the unmatched blocks before asking for a review again.

File: src/test/java/ca/uhn/fhir/jpa/starter/ExampleServerR5IT.java
Comment (hankwallace) [inline src/test/java/ca/uhn/fhir/jpa/starter/ExampleServerR5IT.java:1]: There are also changes in the upstream repo for `src/test/java/ca/uhn/fhir/jpa/starter/ExampleServerR4IT.java` not shown in this pull request.

File: src/test/smoketest/plain_server.http [hunk 12]
```diff
@@ -220,4 +220,4 @@ POST http://{{host}}/fhir/Patient/{{batch_patient_id}}/$validate
         const resourceType = response.body.resourceType;
         client.assert(resourceType === "OperationOutcome", "Expected 'OperationOutcome' but received '" + resourceType + "'");
     });
-%}
\ No newline at end of file
+ %}
```
Comment (hankwallace) [inline src/test/smoketest/plain_server.http]: Why the small difference from the upstream repo?

**Review by hankwallace — CHANGES_REQUESTED** [review]

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java (hunk 9 above)
Comment (shubhamparikh927) [reply src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: I was also thinking importing the specific class instead of `import .*` will update the code same as upstream

**Review by hankwallace — APPROVED** [review]
👍

**Review by vsinghht — APPROVED** [review]
